src/components/Modal/index.js

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { createPortal } from 'react-dom';
+import classnames from 'classnames';
+import Header from './header';
+import RenderIf from '../RenderIf';
+import { uniqueId } from './../../libs/utils';
+import './styles.css';
+
+export default class Modal extends Component {
+    constructor(props) {
+        super(props);
+        this.modalHeadingId = uniqueId('modal-heading');
+        this.modalContentId = uniqueId('modal-content');
+    }
+
+    getContainerClassNames() {
+        const { className } = this.props;
+        return classnames(
+            'rainbow-modal',
+            'rainbow-modal--fade-in',
+            this.getSizeClassNames(),
+            className);
+    }
+
+    getSizeClassNames() {
+        const { size } = this.props;
+        if (size) {
+            return `rainbow-modal--${size}`;
+        }
+        return null;
+    }
+
+    render() {
+        const {
+            title,
+            style,
+            children,
+            footer,
+            isOpen,
+            onRequestClose,
+        } = this.props;
+        return (
+            createPortal(
+                <RenderIf isTrue={isOpen}>
+                    <div className="rainbow-backdrop rainbow-modal--fade-in">
```

Comment:
rainbow-backdrop don't follow the className convention and also the this.getContainerClassNames() should be here and not in section element. Remember that className prop must be applied to container element.